cmd/root.go

```diff
@@ -220,7 +220,15 @@ func initConfigWithViperCtx(c *config) error {
 		}
 	}
 
-	client, err := metalgo.NewDriver(driverURL, apiToken, hmacKey, metalgo.AuthType(hmacAuthType))
+	var (
+		client metalgo.Client
+		err    error
+	)
+	if hmacAuthType != "" {
+		client, err = metalgo.NewDriver(driverURL, apiToken, hmacKey, metalgo.AuthType(hmacAuthType))
+	} else {
+		client, err = metalgo.NewDriver(driverURL, apiToken, hmacKey)
+	}
```

Comment:
You can set default values for viper!

Reply:
You mean like this? Unsure if its better than the if statement, but I can change it, if you like.

```golang
viper.SetDefault("hmac-auth-type", ctx.HMACAuthType)
hmacAuthType := viper.GetString("hmac-auth-type")
```